src/JuliaFormatter.jl

```diff
@@ -358,7 +370,15 @@ function format_text(cst::CSTParser.EXPR, style::AbstractStyle, s::State)
     end
 
     text = String(take!(io))
-    text = normalize_line_ending(text)
+
+    replacer = if s.opts.normalize_line_endings === "unix"
+        WINDOWS_TO_UNIX
+    elseif s.opts.normalize_line_endings === "windows"
+        UNIX_TO_WINDOWS
+    else
+        choose_line_ending_replacer(s.doc.text)
```

Comment:
This is the original text, right? Is there a better way to get that?

Reply:
not atm I think it's only referred to in one other place.

I think using the text arg in the function should work too - you would just have to not override it like we do right now on line 372

Reply:
This function only takes `cst::CSTParser.EXPR, style::AbstractStyle, s::State`, so we don't have that in here :)

Reply:
oops misread that, then yeah s.doc.text is the original text :)